graph/validation_barrier.go

```diff
@@ -180,16 +249,22 @@ func (v *ValidationBarrier) WaitForDependants(job interface{}) error {
 	// Any ChannelUpdate or NodeAnnouncement jobs will need to wait on the
 	// completion of any active ChannelAnnouncement jobs related to them.
 	case *lnwire.ChannelUpdate1:
-		signals, ok = v.chanEdgeDependencies[msg.ShortChannelID]
+		annID = msg.ShortChannelID
+
+		// TODO: If ok is false, we have serious issues.
```

Comment:
throw the error? (if it is really impossible then panic)

Reply:
Despite the many legitimate uses of panics, they have been rejected every time I have tried to use them (even for the provably impossible scenario). I believe that a critical log is the next best thing.

Reply:
returning error now